### PR TITLE
Added a kick command

### LINKED ISSUE #16: Kick Command
We should add a Kick command under Moderation category.

### PR DESCRIPTION
**Pull request by SquarePear**
WIP

I still haven't tested enough for this "powerful" of a command.

Fixes #16

### CODE REVIEW

File: src/commands/kick.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import { sendMessage } from "https://raw.githubusercontent.com/Skillz4Killz/Discordeno/v7/src/handlers/channel.ts";
+
+botCache.commands.set(`kick`, {
+  name: `kick`,
+  guildOnly: true,
+  arguments: [
+    {
+      name: "member",
+      type: "member",
+      required: true,
```
Comment (Skillz4Killz) [inline src/commands/kick.ts:15]: This can be removed as the default is true.

File: src/commands/kick.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      defaultValue: "No reason given",
+    },
+  ],
+  userServerPermissions: [
+    "KICK_MEMBERS",
+  ],
+  botServerPermissions: [
+    "KICK_MEMBERS",
+  ],
+  execute: (message, _args, guild) => {
```
Comment (Skillz4Killz) [inline src/commands/kick.ts:32]:
```suggestion
  execute: (message, args, guild) => {
```

File: src/commands/kick.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    const member: Member = _args["member"];
```
Comment (Skillz4Killz) [inline src/commands/kick.ts:33]: All args accessing should be like this please.
```ts
    const member = args.member;
```

File: src/commands/kick.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    const reason: string = _args["reason"];
+
+    if (!guild || !member || !reason) return;
```
Comment (Skillz4Killz) [inline src/commands/kick.ts:36]: Once you fix the args accessing, you will not need to check any of these. All of these will exist

File: src/commands/kick.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+    return kick(guild.id, member.user.id, reason).then(() => {
```
Comment (Skillz4Killz) [inline src/commands/kick.ts:38]: Although this is fine, it should be done using async/await if possible plz

```ts
try {
await kick(guild.id,,........);

const embed = new Embed()
.....
} catch(error) {
  
return sendMessage(message.channel, "Attempt to kick user has failed");
}
```

File: src/commands/kick.ts [hunk 6]
```diff
@@ -29,13 +28,13 @@ botCache.commands.set(`kick`, {
   botServerPermissions: [
     "KICK_MEMBERS",
   ],
-  execute: (message, _args, guild) => {
-    const member: Member = _args["member"];
-    const reason: string = _args["reason"];
+  execute: async (message, args, guild) => {
+    const member: Member = args.member;
```
Comment (Skillz4Killz) [inline src/commands/kick.ts:32]: This is fine and does work but it goes against the consistency with the rest of the commands. If possible can you do it similar to how the guide showed using something like

```ts
execute: async (message, args: KickArgs, guild) => {

}

interface KickArgs {
  member: Member;
  reason: string;
}
```
Once you do that both these lines can be removed

File: src/commands/kick.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
+import { Embed } from "./../utils/Embed.ts";
+import { Member } from "https://raw.githubusercontent.com/Skillz4Killz/Discordeno/v7/src/structures/member.ts";
```
Comment (Skillz4Killz) [inline src/commands/kick.ts:2]: Use deps.ts

File: src/commands/kick.ts [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  ],
+  userServerPermissions: [
+    "KICK_MEMBERS",
+  ],
+  botServerPermissions: [
+    "KICK_MEMBERS",
+  ],
+  execute: async (message, args: KickArgs, guild) => {
+    try {
+      await kick(guild!.id, args.member.user.id, args.reason);
```
Comment (Skillz4Killz) [inline src/commands/kick.ts:33]:
```suggestion
      await kick(message.guildID, args.member.user.id, args.reason);
```

File: src/commands/kick.ts [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+      const embed = new Embed()
+        .setColor("#FFA500")
+        .setTitle(`Kicked User`)
+        .addField("User:", `<@${args.member.user.id}>`, true)
```
Comment (Skillz4Killz) [inline src/commands/kick.ts:38]:
```suggestion
        .addField("User:", args.member.mention, true)
```

File: src/commands/kick.ts [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+        .addField("Kicked By:", `<@${message.author.id}>`, true)
+        .addField("Reason:", args.reason)
+        .setTimestamp(Date.now());
```
Comment (Skillz4Killz) [inline src/commands/kick.ts:41]:
```suggestion
        .setTimestamp();
```

**Comment by SquarePear** [comment]
I believe this command is ready to be added finally 😄

**Pull request merged**